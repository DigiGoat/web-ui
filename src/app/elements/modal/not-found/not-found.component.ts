import { Component, Input } from '@angular/core';

@Component({
    selector: 'app-modal-not-found',
    templateUrl: './not-found.component.html',
    styleUrl: './not-found.component.scss',
    standalone: false
})
export class NotFoundComponent {
  @Input({ required: true }) searchParam!: string;
}
