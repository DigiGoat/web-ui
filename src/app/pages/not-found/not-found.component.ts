import { Component, type OnDestroy, type OnInit } from '@angular/core';
import { Meta } from '@angular/platform-browser';
import type { Observable } from 'rxjs';
import type { Page } from '../../app-routing.module';

@Component({
    selector: 'app-not-found',
    templateUrl: './not-found.component.html',
    styleUrl: './not-found.component.scss',
    standalone: false
})
export class NotFoundComponent implements OnInit, Page, OnDestroy {
  constructor(private meta: Meta) { }
  ngOnInit(): void {
    this.setDescription();
  }
  setDescription(): void | Observable<void> {
    this.meta.addTag({ name: 'robots', content: 'noindex' });
  }
  ngOnDestroy(): void {
    this.meta.removeTag('name="robots"');
  }
}
