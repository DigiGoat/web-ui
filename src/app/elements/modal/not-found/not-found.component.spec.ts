import { ComponentFixture, TestBed } from '@angular/core/testing';

import { NotFoundComponent } from './not-found.component';

describe('NotFoundComponent', () => {
  let component: NotFoundComponent;
  let fixture: ComponentFixture<NotFoundComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [NotFoundComponent]
    })
      .compileComponents();

    fixture = TestBed.createComponent(NotFoundComponent);
    component = fixture.componentInstance;
    component.searchParam = 'TEST';
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should display a not found message', () => {
    const element = fixture.nativeElement.querySelector('[test-id=message]');
    expect(element.innerHTML).toBe('The Requested Goat "TEST" Was Not Found');
  });
  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
