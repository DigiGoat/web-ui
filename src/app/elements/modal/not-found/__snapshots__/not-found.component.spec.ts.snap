// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent should match the snapshot 1`] = `
<app-modal-not-found
  searchParam={[Function String]}
>
  <h5
    class="card-title text-center"
  >
    Not Found
  </h5><h6
    class="card-subtitle mb-2 text-center"
    test-id="message"
  >
    The Requested Goat "TEST" Was Not Found
  </h6>
</app-modal-not-found>
`;
